torchaudio/functional.py

```diff
@@ -844,6 +844,163 @@ def compute_deltas(specgram, win_length=5, mode="replicate"):
+
+    return waveform * ratio
+
+
+@torch.jit.script
+def scale_to_interval(waveform, interval=1.0):
+    # type: (Tensor, float) -> Tensor
+    r"""Scales waveform to an interval.
+    Args:
+       waveform (torch.Tensor): Tensor of audio of dimension (channel, time)
+       interval (float): The bounds of the interval, where the float indicates
+           the upper bound and the negative of the float indicates the lower
+           bound (Default: `1.0`)
+           Example: interval=1.0 -> [-1.0, 1.0]
+    Returns:
+       torch.Tensor: waveform scaled to interval
+    """
+    abs_max = torch.max(torch.abs(waveform))
+    ratio = abs_max / interval
+    waveform /= ratio
+
+    return waveform
+
+
+def _add_noise_shaping(dithered_waveform, waveform):
+    r"""Noise shaping is calculated by error:
+    error[n] = dithered[n] - original[n]
+    noise_shaped_waveform[n] = dithered[n] + error[n-1]
+    """
+    error = dithered_waveform - waveform
+
+    # add error[n-1] to dithered_waveform[n], so offset the error by 1 index
+    for index in range(error.size()[0]):
+        err = error[index]
+        error_offset = torch.cat((torch.zeros(1), err))
+        error[index] = error_offset[:waveform.size()[1]]
+
+    noise_shaped = dithered_waveform + error
+    return noise_shaped
+
+
+@torch.jit.script
+def probability_distribution(waveform, density_function="TPDF"):
+    # type: (Tensor, str) -> Tensor
+    r"""Perform a probability distribution function on a waveform.
+
+    Triangular probability density function (TPDF) dither noise has a 
+    triangular distribution; values in the center of the range have a higher 
+    probability of occurring. 
+
+    Rectangular probability density function (RPDF) dither noise has a 
+    uniform distribution; any value in the specified range has the same
+    probability of occurring.
+
+    Gaussian probability density function (GPDF) has a normal distribution.
+    The relationship of probabilities of results follows a bell-shaped,
+    or Gaussian curve, typical of dither generated by analog sources.
+    Args:
+        waveform (torch.Tensor): Tensor of audio of dimension (channel, time)
+        probability_density_function (string): The density function of a
+           continuous random variable (Default: `TPDF`)
+           Options: Triangular Probability Density Function - `TPDF`
+                    Rectangular Probability Density Function - `RPDF`
+                    Gaussian Probability Density Function - `GPDF`
+    Returns:
+        torch.Tensor: waveform dithered with TPDF
+    """
+    wave_size = waveform.size()[1] - 1
+    number_of_bits = 16
+
+    up_scaling = 2 ** (number_of_bits - 1) - 2
+    signal_scaled = waveform * up_scaling
+    down_scaling = 2 ** (number_of_bits - 1)
+
+    distributed = waveform
+    if (density_function == "RPDF"):
+        random_tensor = int(torch.randint(wave_size, [1, ]).item())
+        RPDF = waveform[0][random_tensor] - 0.5
```

Comment:
Should this be its own small distribution function to mirror the syntax of the other cases?

Reply:
I could do that, should I make a separate PR for it?

Reply:
Yes, that would be ok in a separate PR.

Reply:
As you pointed out offline, it would be even better to follow what you do on Line 984 in all three cases.
```
        TPDF = torch.bartlett_window(time_size + 1)
```

Reply:
The window functions we have defined are 
- `torch.bartlett_window`
- `torch.blackman_window`
- `torch.hamming_window`
- `torch.hann_window`

So since we don't have one for Rectangular (Dirichlet window) or Gaussian (Gaussian window), should I write both of those?

It looks like a [Hamming window](https://upload.wikimedia.org/wikipedia/commons/thumb/4/4f/Window_function_and_frequency_response_-_Hamming_%28alpha_%3D_0.53836%2C_n_%3D_0...N%29.svg/960px-Window_function_and_frequency_response_-_Hamming_%28alpha_%3D_0.53836%2C_n_%3D_0...N%29.svg.png) could be similar to a [gaussian window](https://upload.wikimedia.org/wikipedia/commons/thumb/0/05/Window_function_and_frequency_response_-_Gaussian_%28sigma_%3D_0.4%29.svg/960px-Window_function_and_frequency_response_-_Gaussian_%28sigma_%3D_0.4%29.svg.png), but they're definitely [defined as different window functions everywhere](https://www.dsprelated.com/freebooks/sasp/Spectrum_Analysis_Windows.html). It DOES however, say that a hamming window is a "Rectangular + scaled-cosine window", so in theory I could device by one period of a cosine.

Reply:
If I write these, should I include them in `torch` or just keep them in `functions.py`?

Reply:
Let's start i in `functional.py` for now. We can migrate them later.

Reply:
As commented [here](https://github.com/pytorch/audio/pull/319#discussion_r346965072), let's punt this for a subsequent PR.